tests/goal.test.sh

```
@@ -368,6 +371,40 @@ assert_grep "$GOAL_SKILL" '_UBERDEV_GOAL_DEFAULT_MAX_PARALLEL=3'         "G24.de
+fi
+# Skip-check matches all three pre-resolved in-flight states (literal alternation in case-arm).
+if grep -qF 'dispatched|solving|pr-pushed) continue' "$GOAL_SKILL"; then
+  PASS=$((PASS+1)); echo "  PASS  G24b.skip-check-extended-to-dispatched"
+else
+  FAIL=$((FAIL+1)); echo "  FAIL  G24b.skip-check-extended-to-dispatched (literal case-arm not found)" >&2
+fi
+# Pre-spawn transition (input->dispatched) appears BEFORE uberdev_dispatch_one.
+# An awk one-shot keeps the line-order check unambiguous: the input->dispatched
+# transition must precede the uberdev_dispatch_one call inside Phase 1.
+_g24b_pre_spawn_line="$(grep -nF 'input dispatched' "$GOAL_SKILL" | head -1 | cut -d: -f1)"
+_g24b_dispatch_call_line="$(grep -nE 'uberdev_dispatch_one "\$ISSUE_NUM"' "$GOAL_SKILL" | head -1 | cut -d: -f1)"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)